tests/test_rpc.py

```diff
@@ -27,58 +27,64 @@
 # LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
 # NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-import unittest
+import pytest
 
-from metricq.rpc import RPCBase, rpc_handler
+from metricq.rpc import RPCDispatcher, rpc_handler
 
 
-class RPCSimple(RPCBase):
+class RPCSimple(RPCDispatcher):
     def __init__(self, number):
         self.number = number
 
     @rpc_handler("test")
-    def handle_test(self):
+    async def handle_test(self):
         return self.number
 
     @rpc_handler("toast")
-    def handle_toast(self, name):
+    async def handle_toast(self, name):
         return self.number * name
 
 
 class RPCSub(RPCSimple):
-    def __init__(self, number):
-        self.number = number
-
     @rpc_handler("sub")
-    def handle_sub(self):
+    async def handle_sub(self):
         return self.number * 2
 
 
-class RPCOther(RPCBase):
+class RPCOther(RPCDispatcher):
     @rpc_handler("foo")
-    def handle_foo(self):
+    async def handle_foo(self):
         return "foo"
 
 
-class TestRPC(unittest.TestCase):
-    def test_basic(self):
-        x = RPCSimple(1)
-        self.assertEqual(x.dispatch("test"), 1)
-        self.assertEqual(x.dispatch("toast", "x"), "x")
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "number, function, kwargs, result",
+    [
+        (1, "test", {}, 1),
+        (2, "test", {}, 2),
+        (1, "toast", {"name": "foo"}, "foo"),
+        (2, "toast", {"name": "foo"}, "foofoo"),
```

Comment:
Renaming the RPCs / Classes here would help readability. E.g. something along the lines of  "echo", "multiply" and "twice".

Reply:
Surely!

Reply:
See 79528c1.